lib/ansible/modules/cloud/amazon/efs_facts.py

```diff
@@ -360,17 +343,15 @@ def main():
     tags = module.params.get('tags')
     targets = module.params.get('targets')
 
-    file_systems_info = connection.get_file_systems(FileSystemId=fs_id, CreationToken=name)
+    file_systems_info = connection.get_file_systems(fs_id, name)
 
     if tags:
-        file_systems_info = filter(lambda item: has_tags(item['Tags'], tags), file_systems_info)
+        file_systems_info = [item for item in file_systems_info if has_tags(item['tags'], tags)]
 
     if targets:
         targets = [(item, prefix_to_attr(item)) for item in targets]
-        file_systems_info = filter(lambda item:
-                                   has_targets(item['MountTargets'], targets), file_systems_info)
+        file_systems_info = [item for item in file_systems_info if has_targets(item['mount_targets'], targets)]
```

Comment:
Can you fix the docs for targets? It's documented as a list of dictionaries, but it needs to just be a list for this to work
```
            targets:
                - "subnet-17bbda4c"
                - "172.31.29.72"
                - [
                    "sg-29d13055",
                    "sg-5e27db2f"
                ]
```
(note that this doesn't work with multiple security groups provided, but that's a preexisting bug and would need has_targets() to be fixed a little).

Reply:
Presumably

```
targets:
                - "subnet-17bbda4c"
                - "172.31.29.72"
                - "sg-29d13055"
                - "sg-5e27db2f"
```

would work though?

Reply:
Docs updated as requested.

Reply:
Ah, that does work. :-)